packages/editor/src/lib/components/LiveCollaborators.tsx

```diff
@@ -33,29 +32,31 @@ const Collaborator = track(function Collaborator({ userId }: { userId: string })
 	const latestPresence = usePresence(userId)
 
 	const [isTimedOut, setIsTimedOut] = useState(false)
+	const rLastSeen = useRef(-1)
 
 	useEffect(() => {
-		// By default, show the cursor
-		setIsTimedOut(false)
-
-		// After a few seconds of inactivity, hide the cursor
-		const timeout = setTimeout(() => {
-			setIsTimedOut(true)
-		}, COLLABORATOR_TIMEOUT)
+		const interval = setInterval(() => {
+			setIsTimedOut(rLastSeen.current - Date.now() > COLLABORATOR_TIMEOUT)
+		}, COLLABORATOR_CHECK_INTERVAL)
 
-		return () => clearTimeout(timeout)
-	}, [latestPresence?.lastActivityTimestamp])
+		return () => clearInterval(interval)
+	}, [])
```

Comment:
lastActivityTimestamp is going to update on every render, including when the cursor is moving, so it's probably too often to update state or swap timeouts around. This PR uses an interval instead to check whether to timeout the user. It will have less accuracy and run even when the user is completely idle, however it will be cheaper to use when the app cursor is moving around.